### PR TITLE
tools/ci: Switch kconfig-frontend to kconfiglib

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

Follow https://github.com/apache/nuttx/pull/8494
kconfiglib can catch more error than kconfig-frontend: https://github.com/apache/nuttx/pull/8485

## Impact

build system

## Testing

CI

### CODE REVIEW

File: tools/ci/docker/linux/Dockerfile [hunk 1]
```diff
@@ -285,13 +268,19 @@ RUN pip3 install cvt2utf
...
 RUN pip3 install imgtool
+RUN pip3 install kconfiglib
 RUN pip3 install pexpect==4.8.0
 RUN pip3 install pyelftools
 RUN pip3 install pyserial==3.5
 RUN pip3 install pytest==6.2.5
 RUN pip3 install pytest-json==0.4.0
 RUN pip3 install pytest-ordering==0.6
 RUN pip3 install pytest-repeat==0.9.1
+# We are using the minimal python installation from the system so include
```
Comment (btashton) [inline tools/ci/docker/linux/Dockerfile:279]: Why did you move this? It is intentionally installed prior to all the other python packages.

**Review by btashton — CHANGES_REQUESTED** [review]

File: tools/ci/docker/linux/Dockerfile (hunk 1 above)
Comment (xiaoxiang781216) [reply tools/ci/docker/linux/Dockerfile:279]: Just want to keep the order:), let me restore it.

Comment (btashton) [reply tools/ci/docker/linux/Dockerfile]: > Just want to keep the order:), let me restore it.

Ah yeah we needed these first to make sure we got wheels in the actual packages we care about. We should probably at some point just move those to be in a requirements.txt to keep them in one place and speed up the build, but that will be a little tricky with relative paths for the Dockerfile.

**Review by anchao — APPROVED** [review]

File: tools/Unix.mk [hunk 2]
```diff
@@ -616,18 +619,19 @@ ifeq ($(KCONFIG_LIB),)
...
+  KCONFIG_SAVEDEFCONFIG = kconfig-conf Kconfig --savedefconfig defconfig.tmp
 define kconfig_tweak_disable
 	kconfig-tweak --file $1 -u $2
 endef
 else
-  KCONFIG_OLDCONFIG     = oldconfig
-  KCONFIG_OLDDEFCONFIG  = olddefconfig
-  KCONFIG_MENUCONFIG    = menuconfig
-  KCONFIG_NCONFIG       = guiconfig
+  PURGE_MODULE_WARNING  = 2> >(grep -v "warning: the 'modules' option is not supported")
```
Comment (xiaoxiang781216) [inline tools/Unix.mk:627]: why redirect 2 to empty?

Comment (anchao) [reply tools/Unix.mk:627]: anonymous pipes :)

https://unix.stackexchange.com/questions/3514/how-to-grep-standard-error-stream-stderr

**Review by GUIDINGLI — APPROVED** [review]

**Comment by davids5** [comment]
@xiaoxiang781216 @anchao - this broke "make menuconfig"  see https://github.com/apache/nuttx/issues/8557#issuecomment-1435126070
